build_tools/third_party/tracy/CMakeLists.txt

```diff
@@ -1,254 +0,0 @@
-# Copyright 2021 The IREE Authors
-#
-# Licensed under the Apache License v2.0 with LLVM Exceptions.
-# See https://llvm.org/LICENSE.txt for license information.
-# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-
-cmake_minimum_required(VERSION 3.16.3)
-
-project(IREETracyServer C CXX)
-
-set(TRACY_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../../third_party/tracy")
-
-find_package(Threads REQUIRED)
-
-#-------------------------------------------------------------------------------
-# Detect package manager
-#-------------------------------------------------------------------------------
-
-message(STATUS "Checking for Tracy dependencies...")
-find_program(PKGCONFIG pkg-config)
-if(NOT PKGCONFIG)
-  message(STATUS "Could not find pkg-config to get dependencies; you need to specify them manually or use pkg-config")
-  message(STATUS "  Ubuntu/Debian: `apt install pkg-config`")
-  message(STATUS "  MacOS: `brew install pkg-config`")
-else()
-  include(FindPkgConfig)
-
-  # Deps slightly differ by platform but some are common.
-  pkg_check_modules(TRACY_DEPS
-    tbb
-    libzstd
-  )
-  pkg_check_modules(TRACY_CAPSTONE_DEPS
-    capstone
-  )
-  if(APPLE)
-    pkg_check_modules(TRACY_GUI_DEPS
-      freetype2
-      glfw3
-    )
-  else()
-    pkg_check_modules(TRACY_GUI_DEPS
-      freetype2
-      glfw3
-      gtk+-3.0
-      dbus-1
-    )
-  endif()
-
-  if(NOT TRACY_DEPS_FOUND)
-    message(STATUS "Could not find Tracy dependencies (Tracy server will not be built).")
-    message(STATUS "To build Tracy, install packages libzstd, and tbb:")
-    message(STATUS "  Ubuntu/Debian: `apt install libcapstone-dev libtbb-dev libzstd-dev`")
-    message(STATUS "  MacOS: `brew install capstone tbb zstd`")
-    return()
-  endif()
-
-  if(NOT TRACY_CAPSTONE_DEPS_FOUND)
-    message(STATUS "Could not find capstone, a Tracy dependency (Tracy server will not be built).")
-    message(STATUS "To build Tracy, install capstone or build from source:")
-    message(STATUS "  Ubuntu/Debian: `apt install libcapstone-dev`")
-    message(STATUS "  MacOS: `brew install capstone`")
-    return()
-  endif()
-
-  if(NOT TRACY_GUI_DEPS_FOUND)
-    message(STATUS "Could not find deps required to build graphical programs (Tracy graphical profiler will not be built).")
-    message(STATUS "  Ubuntu/Debian: `apt install libglfw3-dev libfreetype-dev libgtk-3-dev libdbus-1-dev`")
-    message(STATUS "  MacOS: `brew install glfw freetype`")
-  endif()
```

Comment:
These are still needed, but I'd prefer to direct people to the manual:
![image](https://github.com/iree-org/iree/assets/4010439/f9873eea-5744-4eb3-9065-a9edeb9fbbf4)

Reply:
Some part of the CI (that was working before) wants more deps:
https://github.com/iree-org/iree/actions/runs/9213387061/job/25347181058?pr=17488#step:8:1462 

Going to see if there are CMake options to set that will relax the requirements

Reply:
Can't repro this one locally :/

https://github.com/iree-org/iree/actions/runs/9214146723/job/25349707252?pr=17488#step:7:1830
```
 [255/721] Building CXX object tracy/capture/CMakeFiles/TracyServer.dir/__/server/TracyMmap.cpp.o
  [256/721] Linking C static library tracy/capture/libTracyGetOpt.a
  FAILED: tracy/capture/libTracyGetOpt.a
  : && /tmp/pip-build-env-s3aj7_7v/overlay/lib/python3.9/site-packages/cmake/data/bin/cmake -E rm -f tracy/capture/libTracyGetOpt.a && "CMAKE_C_COMPILER_AR-NOTFOUND" qc tracy/capture/libTracyGetOpt.a  tracy/capture/CMakeFiles/TracyGetOpt.dir/__/getopt/getopt.c.o && "CMAKE_C_COMPILER_RANLIB-NOTFOUND" tracy/capture/libTracyGetOpt.a && :
  /bin/sh: CMAKE_C_COMPILER_AR-NOTFOUND: command not found
```

also found some project-wide config code that we might want to avoid: https://github.com/wolfpld/tracy/blob/afe5c8dfbed060ff0bfe584af028175df6fc4a03/cmake/config.cmake#L45-L50

Might bring back the `build_tools/third_party/tracy/CMakeLists.txt` file 🤔

Reply:
Bah. Decided to just bring back the old file to keep control over global CMake options. I did at least remove the parts we don't strictly need so we can direct IREE developers to use the upstream build for tools that we don't package into our releases.